File: backend/src/main/java/com/doan/backend/entity/Review.java
package com.doan.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Entity
@Table(name = "reviews")
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String id;

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    Product product;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "order_id", nullable = false)
    Order order;

    @Column(name = "rating")
    Double rating;

    @Column(name = "content")
    String content;
}

File: client/src/types/review.ts
export interface ReviewResponse {
    id: string;
    orderId: string;
    productId: string;
    userReviewResponse: {
        id: string;
        name: string;
    };
    rating: number;
    content: string;
}

export interface ReviewRequest {
    productId: string;
    orderId: string;
    rating: number;
    content: string;
}